Add doc for SCIM integration using entity types. (#1366)
FusionAuth/fusionauth-site: site/docs/v1/tech/core-concepts/scim.adoc

- You need to enable SCIM support for your tenant.
- You need to define your incoming request and outgoing response lambdas for each the supported SCIM resource types (User, Enterprise User, Group)
- You will need to verify that the default SCIM schemas provided match your desired SCIM schema for each of the SCIM resource types or provide your own.

> - You will need an Entity defined for each SCIM client and SCIM Server. link:/docs/v1/tech/core-concepts/entity-management#scim-integration[Default entity types] are provided for you. You only need to create the entities from those default types.

Reviewer: ```suggestion
- You will need an Entity defined for each SCIM client and SCIM Server. link:/docs/v1/tech/core-concepts/entity-management#scim-integration[Default entity types] are provided for you. You can create the entities from those default types or create your own types.
```
